### PR TITLE
Resolving issue 6, add highscore to game (#27) fixes #6

### LINKED ISSUE #6: Store the highest streak in localStorage and show it on the right side
We can tell the user what the last highest streak was and so they have an incentive to play the game.

### PR DESCRIPTION
**Pull request by newvertex**
Resolving issue https://github.com/itaditya/got-death-quiz/issues/6

### CODE REVIEW

File: script.js [hunk 1]
```diff
@@ -21,6 +22,24 @@ function incrementScore() {
   return newScore;
 }
 
+function saveHighScore() {
+  if (store.score > store.highScore) {
+    localStorage.setItem('highScore', store.score)
+    return showHighScore();
```
Comment (itaditya) [inline script.js]: don't call showHighScore here

File: script.js [hunk 2]
```diff
@@ -21,6 +22,24 @@ function incrementScore() {
...
+  }
+
+  return 0;
```
Comment (itaditya) [inline script.js]: don't return anything.

Comment (itaditya) [inline script.js]: A better way to write this code is

```js
function saveHighScore() {
  if (store.score <= store.highScore) {
    return;
  }

  localStorage.setItem('highScore', store.score);
}
```

File: script.js [hunk 3]
```diff
@@ -143,10 +162,12 @@ function playGame() {
 
 function endGame() {
   // other stuff to do when game over say saving score will be done here.
+  saveHighScore();
```
Comment (itaditya) [inline script.js:164]:
```suggestion
  saveHighScore();
  showHighScore();
```

File: script.js [hunk 4]
```diff
@@ -21,6 +22,24 @@ function incrementScore() {
...
+  return 0;
+}
+
+function showHighScore() {
+  const newHighScore = localStorage.getItem('highScore') || 0;
+  store.highScore = newHighScore;
+
+  document.querySelector('.js-high-streak').innerText = newHighScore;
+
+  return newHighScore;
```
Comment (itaditya) [inline script.js]: remove this line.

File: style.css [hunk 5]
```diff
@@ -124,6 +124,22 @@ body {
...
+}
+
+.high-score-streak {
+  font-size: 100px;
+  color: #a51717;
+  line-height: 1;
+  font-weight: bold;
+}
+
+.high-score-static-text {
```
Comment (itaditya) [inline style.css]: remove this if its not needed

**Comment by newvertex** [comment]
@itaditya thanks for your advice, I've made changes

File: script.js (hunk 2 above)
Comment (newvertex) [reply script.js]: yeah, this is much better

**Review by itaditya — APPROVED** [review]

**Comment by itaditya** [comment]
Congrats @newvertex!

**Pull request merged**